applications/visit/run.sh

```diff
@@ -0,0 +1,211 @@
+
+  # launch VNC session
+  VNC_DISPLAY=`$VNCSERVER_BIN -rfbauth vncp.txt $@ 2>&1 | grep desktop | awk -F: '{print $3}'`
+  echo "TACC: got VNC display :$VNC_DISPLAY"
+
+  if [ x$VNC_DISPLAY == "x" ]; then
+    echo "TACC: "
+    echo "TACC: ERROR - could not find display created by vncserver: $VNCSERVER"
+    echo "TACC: ERROR - Please submit a consulting ticket at the TACC user portal:"
+    echo "TACC: ERROR - https://portal.tacc.utexas.edu/tacc-consulting/-/consult/tickets/create"
+    echo "TACC: "
+    echo "TACC: job $SLURM_JOB_ID execution finished at: `date`"
+    exit 1
+  fi
+
+  VNC_PORT=`expr 5900 + $VNC_DISPLAY`
+  LOCAL_PORT=5902
+  DISPLAY=":${VNC_DISPLAY}"
+fi
+
+echo "TACC: local (compute node) ${SERVER_TYPE} port is $LOCAL_PORT"
+
+LOGIN_PORT=$(tap_get_port)
+echo "TACC: got login node ${SERVER_TYPE} port $LOGIN_PORT"
+
+# Wait a few seconds for good measure for the job status to update
+sleep 3;
+
+# create reverse tunnel port to login nodes.  Make one tunnel for each login so the user can just connect to $HPC_HOST
+for i in `seq 4`; do
+  ssh -q -f -g -N -R $LOGIN_PORT:$NODE_HOSTNAME:$LOCAL_PORT login$i
+done
+
+echo "TACC: Created reverse ports on $HPC_HOST logins"
+echo "TACC:          https://$HPC_HOST:$LOGIN_PORT"
+
+if [ "x${SERVER_TYPE}" == "xDCV" ]; then
+  INTERACTIVE_SESSION_ADDRESS="https://${HPC_HOST}:${LOGIN_PORT}"
+elif [ "x${SERVER_TYPE}" == "xVNC" ]; then
+
+  TAP_CERTFILE=${HOME}/.tap/.${SLURM_JOB_ID}
+  # bail if we cannot create a secure session
+  if [ ! -f ${TAP_CERTFILE} ]; then
+    echo "TACC: ERROR - could not find TLS cert for secure session"
+    echo "TACC: job ${SLURM_JOB_ID} execution finished at: $(date)"
+    exit 1
+  fi
+
+  # fire up websockify to turn the vnc session connection into a websocket connection
+  WEBSOCKIFY_CMD="/home1/00832/envision/websockify/run"
+  WEBSOCKIFY_PORT=5902
+  WEBSOCKIFY_ARGS="--cert=$(cat ${TAP_CERTFILE}) --ssl-only --ssl-version=tlsv1_2 -D ${WEBSOCKIFY_PORT} localhost:${VNC_PORT}"
+  ${WEBSOCKIFY_CMD} ${WEBSOCKIFY_ARGS} # websockify will daemonize
+
+  INTERACTIVE_SESSION_ADDRESS="https://tap.tacc.utexas.edu/noVNC/?host=${HPC_HOST}&port=${LOGIN_PORT}&password=${TAPIS_JOB_ID}&autoconnect=true&encrypt=true&resize=scale"
+else
+  # we should never get this message since we just checked this at LOCAL_PORT
+  echo "TACC: "
+  echo "TACC: ERROR - unknown server type '${SERVER_TYPE}'"
+  echo "TACC: Please submit a consulting ticket at the TACC user portal"
+  echo "TACC: https://portal.tacc.utexas.edu/tacc-consulting/-/consult/tickets/create"
+  echo "TACC:"
+  echo "TACC: job $SLURM_JOB_ID execution finished at: `date`"
+  exit 1
+fi
+
+# Webhook callback url for job ready notification.
+# Notification is sent to _INTERACTIVE_WEBHOOK_URL, e.g. https://3dem.org/webhooks/interactive/
+curl -k --data "event_type=WEB&address=${INTERACTIVE_SESSION_ADDRESS}&owner=${TAPIS_JOB_OWNER}&job_uuid=${TAPIS_JOB_ID}" $INTERACTIVE_WEBHOOK_URL &
```

Comment:
This will look a bit different in v3
```suggestion
curl -k --data "event_type=interactive_session_ready&address=${INTERACTIVE_SESSION_ADDRESS}&owner=${_tapisJobOwner}&job_uuid=${_tapisJobUUID}" "${_INTERACTIVE_WEBHOOK_URL}" &

```